java/server/test/org/openqa/selenium/grid/router/ProxyCdpTest.java

```diff
@@ -126,9 +132,40 @@ public void onText(CharSequence data) {
     socket.close();
   }
 
-  private Server<?> createBackendServer(CountDownLatch latch, AtomicReference<String> incomingRef, String response) {
+  @Test
+  public void shouldBeAbleToSendMessagesOverSecureWebSocket()
+      throws URISyntaxException, InterruptedException {
+
+    HttpClient.Factory clientFactory = HttpClient.Factory.createDefault();
+
+    Server<?> backend = createBackendServer(new CountDownLatch(1), new AtomicReference<>(), "Cheedar", secureConfig);
+
+    // Push a session that resolves to the backend server into the session map
+    SessionId id = new SessionId(UUID.randomUUID());
+    sessions.add(new Session(id, backend.getUrl().toURI(), new ImmutableCapabilities()));
+
+    CountDownLatch latch = new CountDownLatch(1);
+    AtomicReference<String> text = new AtomicReference<>();
+    try (WebSocket socket = clientFactory.createClient(secureProxyServer.getUrl())
+      .openSocket(new HttpRequest(GET, String.format("/session/%s/cdp", id)), new WebSocket.Listener() {
+        @Override
+        public void onText(CharSequence data) {
+          text.set(data.toString());
+          latch.countDown();
+        }
+      })) {
+      socket.sendText("Cheese!");
+
+      assertThat(latch.await(5, SECONDS)).isTrue();
+      assertThat(text.get()).isEqualTo("Cheedar");
+    } catch (Exception e) {
```

Comment:
This `catch` can be removed if we are going to throw it again. (It will be thrown anyway if the `catch` is removed).